### PR TITLE
Vector pointing mode

### PR DESCRIPTION
**Pull request by X52p**
Implement Vector Pointing mode and fix some problems with the simulator

### CODE REVIEW

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/VectorPointingSimulator.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/*
```
Comment (dmarszk) [inline mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/VectorPointingSimulator.java:1]: Add NMF license header

**Comment by dmarszk** [comment]
Question - would merging it into dev break nmf-ops-sat builds?

File: core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml [hunk 2]
```diff
@@ -1395,6 +1395,25 @@
                       name="AttitudeMode" service="AutonomousADCS" />
           </mal:extends>
         </mal:composite>
+        <mal:composite
+          comment="TODO."
```
[the remark this answers is not in the document]
Comment (X52p) [reply core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml]: done

File: core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml [hunk 3]
```diff
@@ -1395,6 +1395,25 @@
...
+          name="AttitudeModeVectorPointing" shortFormPart="12">
+          <mal:extends>
+            <mal:type area="Platform" list="false"
+                      name="AttitudeMode" service="AutonomousADCS" />
+          </mal:extends>
+          <mal:field canBeNull="false"
+                     comment="TODO"
```
[the remark this answers is not in the document]
Comment (X52p) [reply core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml]: done

File: core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml [hunk 4]
```diff
@@ -1395,6 +1395,25 @@
...
+                     name="target">
+            <mal:type area="Platform" list="false"
+                      name="VectorF3D"/>
+          </mal:field>
+          <mal:field canBeNull="false"
+                     comment="TODO"
```
[the remark this answers is not in the document]
Comment (X52p) [reply core/mo-services-xml/src/main/resources/xml/ServiceDefPLATFORM.xml]: done

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/AttitudeProviderWrapper.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/*
```
[the remark this answers is not in the document]
Comment (X52p) [reply mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/AttitudeProviderWrapper.java:1]: done

File: mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/VectorPointingSimulator.java (hunk 1 above)
Comment (X52p) [reply mission/simulator/opssat-spacecraft-simulator/src/main/java/opssat/simulator/orekit/VectorPointingSimulator.java:1]: done

**Comment by X52p** [comment]
> Question - would merging it into dev break nmf-ops-sat builds?

I tried it, and it does build without errors. I also created a pull request for the corresponding nmf-ops-sat branch. (I also rebased it onto the current state of dev branch)

**Comment by dmarszk** [comment]
@X52p please let me know once this is ready to be reviewed and merged.

**Pull request merged**